src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7418,4 +7430,43 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="WRN_ParamsArrayInLambdaOnly_Title" xml:space="preserve">
     <value>Parameter has params modifier in lambda but not in target delegate type.</value>
   </data>
+  <data name="ERR_BadExtensionUnderlyingType" xml:space="preserve">
+    <value>Extension '{0}' cannot augment underlying type '{1}'. The extension underlying type may not be dynamic, a pointer, a nullable reference type, a ref struct or an extension.</value>
```

Comment:
>Extension '{0}' cannot augment underlying type '{1}'. [](http://example.com/codeflow?start=11&length=53)

I think this sounds too complicated. I would drop the first sentence completely. The second sentence in combination with location should be good enough.